File: VB/WindowsApplication1/CustomControls/CustomCheckedListBoxControl/CustomCheckedListBoxControl.vb
Imports System.ComponentModel
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.ViewInfo
Imports System

Namespace Example
	<ToolboxItem(True)>
	Public Class CustomCheckedListBoxControl
		Inherits CheckedListBoxControl

		Protected Overrides Function CreateViewInfo() As BaseStyleControlViewInfo
			Return New CustomCheckedListBoxViewInfo(Me)
		End Function

		#Region "ShiftItem"
		Public Event ShiftItem As EventHandler(Of ItemInfoEventArgs)
		Protected Friend Sub RaiseShiftItem(ByVal ea As ItemInfoEventArgs)
            RaiseEvent ShiftItem(Me, ea)
		End Sub
		#End Region
	End Class
End Namespace

File: CS/WindowsApplication1/CustomControls/CustomCheckedListBoxControl/CustomCheckedListBoxControl.cs
using System.ComponentModel;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.ViewInfo;
using System;

namespace Example
{
    [ToolboxItem(true)]
    public class CustomCheckedListBoxControl : CheckedListBoxControl
    {
        protected override BaseStyleControlViewInfo CreateViewInfo() { return new CustomCheckedListBoxViewInfo(this); }

        #region ShiftItem
        public event EventHandler<ItemInfoEventArgs> ShiftItem;
        protected internal void RaiseShiftItem(ItemInfoEventArgs ea)
        {
            EventHandler<ItemInfoEventArgs> handler = ShiftItem;
            if (handler != null)
                handler(this, ea);
        }
        #endregion
    }
}
